xml/System.Reflection/AssemblyVersionAttribute.xml

```diff
@@ -86,12 +86,32 @@
   
 > [!IMPORTANT]
 >  All components of the version must be integers greater than or equal to 0. Metadata restricts the major, minor, build, and revision components for an assembly to a maximum value of <xref:System.UInt16.MaxValue?displayProperty=nameWithType> - 1. If a component exceeds this value, a compilation error occurs.  
-  
- You can specify all the values or you can accept the default build number, revision number, or both by using an asterisk (\*). For example, `[assembly:AssemblyVersion("2.3.25.1")]` indicates 2 as the major version, 3 as the minor version, 25 as the build number, and 1 as the revision number. A version number such as `[assembly:AssemblyVersion("1.2.*")]` specifies 1 as the major version, 2 as the minor version, and accepts the default build and revision numbers. A version number such as `[assembly:AssemblyVersion("1.2.15.*")]` specifies 1 as the major version, 2 as the minor version, 15 as the build number, and accepts the default revision number. The default build number increments daily. The default revision number is the number of seconds since midnight local time (without taking into account time zone adjustments for daylight saving time), divided by 2.  
-  
-> [!NOTE]
->  If you specify an asterisk for the build number, you cannot specify a revision number.  
-  
+
+ For example, `[assembly:AssemblyVersion("2.3.25.1")]` indicates 2 as the major version, 3 as the minor version, 25 as the build number, and 1 as the revision number. 
+
+ The <xref:System.Reflection.AssemblyVersionAttribute> attribute allows to specify an asterisk (\*) in place of the build or revision number. A version number such as `[assembly:AssemblyVersion("1.2.*")]` specifies 1 as the major version, 2 as the minor version, and accepts the default build and revision numbers. A version number such as `[assembly:AssemblyVersion("1.2.15.*")]` specifies 1 as the major version, 2 as the minor version, 15 as the build number, and accepts the default revision number. The default build number increments daily. The default revision number is the number of seconds since midnight local time (without taking into account time zone adjustments for daylight saving time), divided by 2. If you specify an asterisk for the build number, you cannot specify a revision number.  
```

Comment:
I didn't even know about this feature.  It's not a runtime feature - the attribute itself is stored as a string.  It must be in the compiler:
https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK7wCYgNQB8ACATAIwCwAUPgAwAE+xAdAEoCmAZgDYsDGCUA9jADcFANoQAzhJZhgHAJ4gaAQSky58gGotkEgTAAUAIkZUGAKiMBKALoV6ATgMwWAdxVrZC7bv3KECMhQwKgILMamFtYMPnqCViKUxE6q0l7yDADiLAgAogAePKGwAOap6goGVlk5AHIQYOHVsfoJFEA==

https://github.com/dotnet/roslyn/blob/18614bceb9d0a9eda5cfb47409c3058971cc6751/src/Compilers/Core/Portable/VersionHelper.cs#L170

Wow, that's indeed broken.  Did we consider an diagnostic to discourage this?